src/ScalarComparator.php

```diff
@@ -51,7 +51,7 @@ public function assertEquals($expected, $actual, $delta = 0.0, $canonicalize = f
 
         // always compare as strings to avoid strange behaviour
```

Comment:
This behaviour exist since 9 years, changing it may be a BC break.

Reply:
that is true, but `0` equals `false`, so `'0'` must too, it is also the php behaviour of weak comparison